### ISSUE TITLE
Upload progress bar for balena push

### ISSUE THREAD

**Issue by zrzka** [balena-io/balena-cli#1303]
It's a nice to have to RFE. For my current project, I'm uploading 2GB of data during `balena push`. There're some spinning characters only and it would be nice to have some upload progress bar. It takes _ages_ and one can be puzzled if it's still doing anything, etc.

**Comment by pdcastro** [comment]
Indeed. And for the case of scriptable / JSON output (#1089) a JSON stream with completion percentage could be produced (like the Docker daemon does with image push/pull).
